subprojects/core/src/test/groovy/org/gradle/api/internal/file/archive/AbstractArchiveFileTreeTest.java

```diff
@@ -54,7 +55,7 @@ public abstract class AbstractArchiveFileTreeTest {
     protected final TestFile rootDir = tempDirProvider.getTestDirectory().file("root");
 
     protected abstract TestFile getArchiveFile();
-    protected abstract AbstractArchiveFileTree getTree();
+    protected abstract FileTree getTree();
```

Comment:
The reason this changed is that the underlying type from `Project.zipTree` or `Project.tarTree` is not a `ZipFileTree` or `TarFileTree`.  It's wrapped in a `FileTreeAdapter` that does some other things. 

I did this to avoid having to deal with the `ZipFileTree` or `TarFileTree` constructors directly and could create them through the "normal" methods a user would use, but this doesn't seem necessary now. At some point, I think I thought we'd need to inject more services that would be hard to stub. 

I'll put this back.

Reply:
I see what you mean.  I still think this way is clearer though, I did briefly wonder where the `AbstractArchiveFileTree` was when reading this for the first time.